proto/skaffold.proto

```diff
@@ -302,53 +317,87 @@ enum ClusterType {
     OTHER = 3;
 }
 
-// Enum for error codes
-enum ErrorCode {
-    // Could not determine error
-    COULD_NOT_DETERMINE = 0;
+// Enum for Status codes
+// These error codes are prepended by Phase Name e.g.
+// BUILD, DEPLOY, STATUSCHECK, DEVINIT
+enum StatusCode {
+    // Could not determine error and phase
+    UNKNOWN_ERROR = 0;
 
     // No Error codes
 
     // Status Check Success
-    STATUS_CHECK_NO_ERROR = 200;
+    STATUSCHECK_SUCCESS = 200;
 
 
     // Container errors
 
     // Container image pull error
-    STATUS_CHECK_IMAGE_PULL_ERR = 300;
+    STATUSCHECK_IMAGE_PULL_ERR = 300;
     // Container creating error
-    STATUS_CHECK_CONTAINER_CREATING = 301;
+    STATUSCHECK_CONTAINER_CREATING = 301;
     // Container run error
-    STATUS_CHECK_RUN_CONTAINER_ERR = 302;
+    STATUSCHECK_RUN_CONTAINER_ERR = 302;
     // Container is already terminated
-    STATUS_CHECK_CONTAINER_TERMINATED = 303;
+    STATUSCHECK_CONTAINER_TERMINATED = 303;
     // Container restarting error
-    STATUS_CHECK_CONTAINER_RESTARTING = 356;
+    STATUSCHECK_CONTAINER_RESTARTING = 356;
 
     // K8 infra errors
 
     // Node memory pressure error
-    STATUS_CHECK_NODE_MEMORY_PRESSURE = 400;
+    STATUSCHECK_NODE_MEMORY_PRESSURE = 400;
     // Node disk pressure error
-    STATUS_CHECK_NODE_DISK_PRESSURE = 401;
+    STATUSCHECK_NODE_DISK_PRESSURE = 401;
     // Node network unavailable error
-    STATUS_CHECK_NODE_NETWORK_UNAVAILABLE = 402;
+    STATUSCHECK_NODE_NETWORK_UNAVAILABLE = 402;
     // Node PID pressure error
-    STATUS_CHECK_NODE_PID_PRESSURE = 403;
+    STATUSCHECK_NODE_PID_PRESSURE = 403;
     // Node unschedulable error
-    STATUS_CHECK_NODE_UNSCHEDULABLE= 404;
+    STATUSCHECK_NODE_UNSCHEDULABLE= 404;
     // Node unreachable error
-    STATUS_CHECK_NODE_UNREACHABLE = 405;
+    STATUSCHECK_NODE_UNREACHABLE = 405;
     // Node not ready error
-    STATUS_CHECK_NODE_NOT_READY= 406;
+    STATUSCHECK_NODE_NOT_READY= 406;
 
     // Unknown Error Codes
 
     // Status Check error unknown
-    STATUS_CHECK_UNKNOWN = 501;
+    STATUSCHECK_UNKNOWN = 501;
     // Container is unschedulable due to unknown reasons
-    STATUS_CHECK_UNKNOWN_UNSCHEDULABLE = 502;
+    STATUSCHECK_UNKNOWN_UNSCHEDULABLE = 502;
     // Container is waiting due to unknown reason
-    STATUS_CHECK_CONTAINER_WAITING_UNKNOWN = 503;
-}
\ No newline at end of file
+    STATUSCHECK_CONTAINER_WAITING_UNKNOWN = 503;
+    // Deploy failed due to unknown reason
+    DEPLOY_UNKNOWN = 504;
+    // SYNC failed due to known reason
+    SYNC_UNKNOWN = 505;
+    // Build failed due to unknown reason
+    BUILD_UNKNOWN= 506;
+    // Dev Init failed due to unknown reason
+    DEVINIT_UNKNOWN = 507;
+    // Cleanup failed due to unknown reason
+    CLEANUP_UNKNOWN = 508;
+
+
+
+
+
+    // FILE_SYNC Failures
+
+    // File Sync Initialize failure
+    SYNC_INIT_ERROR = 601;
```

Comment:
This name seems slightly odd given the DEVINIT_ one below

Reply:
so, `Sync` represents the phase. and `INIT_ERROR` represents the error code. 
This error maps to initialization the sync file watcher. 
There are other possible errors in sync phase like failed to copy a file in container.